### PR TITLE
Threaded tasks for tile building in editor mode

### PR DESCRIPTION
**Pull request by karimnaaji**
- Run tasks asynchronously in editor mode
- Refactor tile task and apply scene groups add the end of the tile building pipeline

### CODE REVIEW

**Comment by matteblair** [comment]
The "running asynchronously" part seems to be working fine, but I'm confused by the scene group building process. It's pretty complex, and there seems to be a lot of work happening in `GenerateSceneGraph()` after all the individual tile tasks are complete. 

I'll give more concrete thoughts after I've thought more about how this might work differently.

**Comment by karimnaaji** [comment]
Yes, sorry I had a commit I intended to push yesterday, I added more comments to it this morning for clarity. Give it another look so I can rebase my other branch with dynamic reloading.

File: Assets/Mapzen/Unity/Editor/RegionMapEditor.cs [hunk 1]
```diff
@@ -53,14 +53,25 @@ public override void OnInspectorGUI()
...
                 else
                 {
                     LogErrors();
                 }
             }
 
+            if (map.HasPendingTasks())
+            {
+                // Go through another OnInspectorGUI cycle
+                Repaint();
```
Comment (matteblair) [inline Assets/Mapzen/Unity/Editor/RegionMapEditor.cs]: So if I understand this part, it's causing continuous `Repaint()`s in a loop while the tile data is building on a background thread? If so, that seems like a lot of CPU time spent doing unneeded rendering. I suppose it does technically keep the UI interactive though. Is there no way to trigger an event on the main thread when the background threads complete their work?

Comment (matteblair) [reply Assets/Mapzen/Unity/Editor/RegionMapEditor.cs]: I'm starting to understand why this is tricky - the background threads can't safely call into the Unity framework to trigger the subsequent tasks :\

I did get this working with a Coroutine started from `RegionMap`. The problem with this approach is that in the editor, Couroutines don't seem to update regularly like they do in a player - which makes sense given that the editor doesn't run in a continuous loop.

I'm ok with merging as-is now, seeing as it's not easy to work around this.

Comment (karimnaaji) [reply Assets/Mapzen/Unity/Editor/RegionMapEditor.cs]: Yes, I'll give a look again into an alternative today and let you know if I find anything. I also tried the Coroutine way and figured the same.

**Pull request merged**